engines/private/private.h

```diff
@@ -0,0 +1,288 @@
+    Common::String *sound;
+    Symbol *flag;
+    int val;
+} PhoneInfo;
+
+typedef struct DossierInfo {
+    Common::String *page1;
+    Common::String *page2;
+} DossierInfo;
+
+// lists
+
+typedef Common::List<ExitInfo> ExitList;
+typedef Common::List<MaskInfo> MaskList;
+typedef Common::List<Common::String> SoundList;
+typedef Common::List<PhoneInfo> PhoneList;
+typedef Common::List<Common::String> InvList;
+
+// arrays
+
+typedef Common::Array<DossierInfo> DossierArray;
+
+// hash tables
+
+typedef Common::HashMap<Common::String, bool> PlayedMediaTable;
+
+class PrivateEngine : public Engine {
+private:
+    Common::RandomSource *_rnd;
+
+    Graphics::PixelFormat _pixelFormat;
+    Image::ImageDecoder *_image;
+    int _screenW, _screenH;
+
+public:
+    PrivateEngine(OSystem *syst, const ADGameDescription *gd);
+    ~PrivateEngine();
+
+    const ADGameDescription *_gameDescription;
+    bool isDemo() const;
+
+    Audio::SoundHandle _fgSoundHandle;
+    Audio::SoundHandle _bgSoundHandle;
+    Video::SmackerDecoder *_videoDecoder;
+    Common::InstallShieldV3 _installerArchive;
+
+    Common::Error run() override;
+    void restartGame();
+    void clearAreas();
+    void initializePath(const Common::FSNode &gamePath) override;
+
+    // User input
+    void selectPauseMovie(Common::Point);
+    void selectMask(Common::Point);
+    void selectExit(Common::Point);
+    void selectLoadGame(Common::Point);
+    void selectSaveGame(Common::Point);
+
+    // Cursors
+    bool cursorPauseMovie(Common::Point);
+    bool cursorExit(Common::Point);
+    bool cursorMask(Common::Point);
+
+    bool hasFeature(EngineFeature f) const override;
+    bool canLoadGameStateCurrently() override {
+        return true;
+    }
+    bool canSaveAutosaveCurrently() override  {
+        return false;
+    }
+    bool canSaveGameStateCurrently() override {
+        return true;
+    }
+
+    Common::Error loadGameStream(Common::SeekableReadStream *stream) override;
+    Common::Error saveGameStream(Common::WriteStream *stream, bool isAutosave = false) override;
+    void syncGameStream(Common::Serializer &s);
+
+    Common::String convertPath(Common::String);
+    void playVideo(const Common::String &);
+    void skipVideo();
+
+    void loadImage(const Common::String &file, int x, int y);
+    void drawScreenFrame(Graphics::Surface *);
+
+    void changeCursor(Common::String);
+    void initCursors();
+
+    // Rendering
+    Graphics::ManagedSurface *_compositeSurface;
+    Graphics::ManagedSurface *loadMask(const Common::String &, int, int, bool);
+    void drawMask(Graphics::ManagedSurface *);
+    bool inMask(Graphics::ManagedSurface *, Common::Point);
+    uint32 _transparentColor;
+    Common::Rect *screenRect;
+    Common::String *_framePath;
+    Graphics::Surface *_frame;
+    Common::String *_nextVS;
+    void drawScreen();
+
+    // global state
+    const Common::Point *_origin;
+    void setOrigin(const int[2]);
+    Common::String *_nextSetting;
```

Comment:
`_nextSetting` pointer is leaked - better would be to use `Common::String _nextSetting` here

Reply:
The same is true for `_currentSetting` and `_pauseSetting` - they might also be leaked in some situations.

Reply:
I think these are fixed. Please check.